extensions/MozChangeField/lib/Post/RegressedByTypeKeyword.pm

```diff
@@ -0,0 +1,87 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::MozChangeField::Post::RegressedByTypeKeyword;
+
+use 5.10.1;
+use Moo;
+
+use Bugzilla::Keyword;
+
+sub evaluate_create {
+  my ($self, $args) = @_;
+  my $bug       = $args->{bug};
+  my $timestamp = $args->{timestamp};
+  my $dbh       = Bugzilla->dbh;
+
+  return unless @{$bug->regressed_by};
```

Comment:
```suggestion
  return unless (@{$bug->regressed_by} || $bug->has_keyword('regression'));
```
In this way, the bug type will be enforced to be `defect` even if the `regressed_by` field is empty.

We could do a similar thing for `evaluate_change`.

Reply:
Thanks. good point